src/tests/multihost/alltests/conftest.py

```diff
@@ -559,6 +559,62 @@ def del_sudo_rule():
 # ====================  Class Scoped Fixtures ================
 
 
+@pytest.fixture(scope='class')
+def create_custom_posix_usersgroups(session_multihost, setup_sssd_krb, create_posix_usersgroups, krb_connection_timeout):
+    """ Create posix user and groups """
+
+    ldap_uri = 'ldap://%s' % (session_multihost.master[0].sys_hostname)
+    ds_rootdn = 'cn=Directory Manager'
+    ds_rootpw = 'Secret123'
+    ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+    krb = krb5srv(session_multihost.master[0], 'EXAMPLE.TEST')
+
+    for i in range(3, 5):
+        user_info = {
+            'cn': 'Test User%d' % i,
+            'uid': 'testuser%d' % i,
+            'uidNumber': '200%d' % i,
+            'gidNUmber': '200%d' % i,
+        }
+        if ldap_inst.posix_user("ou=People", "dc=example,dc=com", user_info):
+            krb.add_principal('testuser%d' % i, 'user', 'Secret123')
+        else:
+            print("Unable to add ldap User %s" % (user_info))
+            assert False
+
+        group_info = {'cn': 'testgroup%d' % i,
+                      'gidNumber': '200%d' % i,
+                      'memberUid': 'testuser%d' % i}
+        try:
+            ldap_inst.posix_group("ou=Groups", "dc=example,dc=com", group_info, True)
+        except LdapException:
+            assert False
+
+        time.sleep(5)
+
+    sssdTools(session_multihost.client[0]).clear_sssd_cache()
+
+    lookups = ["getent -s sss passwd testuser3", "getent -s sss passwd testuser4", "auth_success testuser3 Secret123",
+               "auth_success testuser4 Secret123"]
+    for lookup in lookups:
```

Comment:
Please unwrap the loop (and possibly  add asserts showing which command exactly failed).
As You do not have raiseonerr=False in run_command my guess is that the failed lookup will trigger exception before the assert is called anyway so You might just run the commands.